src/bika/lims/content/worksheet.py

```diff
@@ -440,7 +440,7 @@ def nextRefAnalysesGroupID(self, reference):
             prefix = reference.id + "-D"
         bac = getToolByName(reference, 'senaite_catalog_analysis')
         ids = bac.Indexes['getReferenceAnalysesGroupID'].uniqueValues()
-        rr = re.compile("^" + prefix + "[\d+]+$")
+        rr = re.compile("^" + prefix + "[\\d+]+$")
```

Comment:
An alternative would be to use a raw string as suggested in https://docs.python.org/2/library/re.html:

```suggestion
        rr = re.compile("^" + prefix + r"[\d+]+$")
```

Reply:
Yes, I would prefer the raw strings instead. Would you mind to change it?

Reply:
sure, no problem! :+1:

Reply:
@ramonski I have replaced them with raw strings now :heavy_check_mark: